catalog/solr/catalog-solr-provider/src/main/resources/OSGI-INF/blueprint/blueprint.xml

```diff
@@ -18,6 +18,12 @@
 
 	<reference id="filterAdapter" interface="ddf.catalog.filter.FilterAdapter"/>
 
+	<bean id="dynamicSchemaResolver" class="ddf.catalog.source.solr.DynamicSchemaResolver"/>
+
+	<reference-list id="metacardTypeList" interface="ddf.catalog.data.MetacardType">
+		<reference-listener bind-method="addMetacardType" unbind-method="removeMetacardType" ref="dynamicSchemaResolver"/>
```

Comment:
If you take my previous advice, you should be able to remove `unbind-method`, the `removeMetacardType` method, and the counting map.